### PR TITLE
Add the location detail and the character card

### PR DESCRIPTION
**Pull request by SSPF20**
Simulator Screen Recording - iPhone 14 Pro - 2023-09-07 at 16 17 17

### CODE REVIEW

File: RickAndMorty/Views/DetailViews/RMLocationDetail/RMCharacterCardView.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    
+    var viewModel: RMCharacterCardViewModel
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            VStack() {
+                WebImage(url: viewModel.imageURL)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                ZStack() {
```
Comment (hgp691) [inline RickAndMorty/Views/DetailViews/RMLocationDetail/RMCharacterCardView.swift]: Why we need a ZStack for only 1 entity?

File: RickAndMorty/Views/DetailViews/RMLocationDetail/RMCharacterCardViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    
+    func loadCharacter () {
+        Task { @MainActor in
+            do {
+                let endpoint = RMCharacterEndpoint.characterById(id)
+                let character: RMCharacter = try await networkFetcher.fetch(endpoint: endpoint)
+                self.character = character
+            } catch {
+                assert(false, String(describing: error))
```
Comment (hgp691) [inline RickAndMorty/Views/DetailViews/RMLocationDetail/RMCharacterCardViewModel.swift]: Add a TODO to make the Error handling

File: RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailView.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        ScrollView(.vertical) {
+            VStack {
+                NameValueField(name: "Name: ", value: viewModel.name, font: .title2)
+                NameValueField(name: "Type: ", value: viewModel.type, font: .title2)
+                NameValueField(name: "Dimension: ", value: viewModel.dimension, font: .title2)
+                Spacer()
+            }
+            .padding([.leading, .top])
+            VStack {
+                Rectangle()
```
Comment (hgp691) [inline RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailView.swift]: Replace the Rectangle, add a GIF for clarification

File: RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailView.swift [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        .padding(.top, 1)
+    }
+}
+
+#Preview {
+    RMLocationDetailView(viewModel: .init(location: .init(id: 1,
+                                                          name: "Earth",
+                                                          type: "Planet",
+                                                          dimension: "Dimension C-137",
+                                                          residents: [])))
```
Comment (hgp691) [inline RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailView.swift]: It is worth to add Residents to the Preview ?

File: RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+//  Created by Horacio Parra Rodriguez on 1/09/23.
+//
+
+import Foundation
+import Observation
+
+@Observable
+final class RMLocationDetailViewModel {
+    
+    let location: RMLocation
```
Comment (hgp691) [inline RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailViewModel.swift]: this should  be private in order to isolate the direct access to the Location

File: RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailView.swift (hunk 4 above)
Comment (SSPF20) [reply RickAndMorty/Views/DetailViews/RMLocationDetail/RMLocationDetailView.swift]: I set a default value in the variable

**Review by hgp691 — APPROVED** [review]

**Pull request merged**